gradlew

```diff
@@ -66,8 +66,11 @@ esac
 
 CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar
 
-# Determine the Java command to use to start the JVM.
-if [ -n "$JAVA_HOME" ] ; then
+# Determine the Java command to use to start the JVM.  Use java alongside compileJavac if specified, else use JAVA_HOME, else use PATH
+for ARG; do if echo "$ARG" | grep -q '^-PcompileJavac='; then JAVAC="$(echo $ARG|sed s/^-PcompileJavac=//)"; JAVACMD="$(echo $JAVAC|sed s/javac/java/g)"; fi; done
```

Comment:
I have mixed feelings about this and would like to hear other opinions...

On the one hand: I agree that restricting `JAVA_HOME` will provide a great deal of safety from accidentally leaking java versions where we don't mean to do so.

On the other hand: the `gradlew` wrapper is typically generated with the `gradle wrapper` command, and it feels off to deviate from that in a way that isn't in our `build.gradle`.

On the other other hand: `gradlew` is part of our source tree, so it's not as if we're encouraging developers to go fetch their own gradle wrapper in any event.

I think I'm still +1 on this PR overall, though.  Just a bit conflicted about it.